src/main/java/io/jenkins/plugins/appcenter/task/internal/UploadAppToResourceTask.java

```diff
@@ -84,11 +91,54 @@
         final String pathToDebugSymbols = request.pathToDebugSymbols;
         final String symbolUploadUrl = requireNonNull(request.symbolUploadUrl, "symbolUploadUrl cannot be null");
 
-        log("Uploading symbols to resource.");
+        final File file = new File(filePath.child(pathToDebugSymbols).getRemote());
+        if (file.length() > (1024 * 1024) * 256) {
+            return uploadSymbolsChunked(request, symbolUploadUrl, file);
+        } else {
+            return uploadSymbolsComplete(request, symbolUploadUrl, file);
+        }
+    }
+
+    @Nonnull
+    private CompletableFuture<UploadRequest> uploadSymbolsChunked(@Nonnull UploadRequest request, @Nonnull String symbolUploadUrl, @Nonnull File file) {
+        log("Uploading symbols to resource chunked.");
 
         final CompletableFuture<UploadRequest> future = new CompletableFuture<>();
+        CompletableFuture.supplyAsync(() ->
+        {
+            try {
+                //Workaround for bug in Azure Blob Storage, as AppCenter returns the upload URL with a port attached
+                //See https://github.com/Azure/azure-sdk-for-java/issues/15827
+                final URL oldURL = new URL(symbolUploadUrl);
+                final URL newURL = new URL(oldURL.getProtocol(), oldURL.getHost(), oldURL.getFile());
+                final String symbolUploadUrlWithoutPort = newURL.toString();
+
+                final BlobClient blobClient = new BlobClientBuilder().endpoint(symbolUploadUrlWithoutPort).buildClient();
+                blobClient.uploadFromFile(file.getPath(), true);
+            } catch (Exception e) {
+                final AppCenterException exception = logFailure("Upload symbols to resource unsuccessful: ", e);
+                future.completeExceptionally(exception);
+            }
+            return (Void) null;
```

Comment:
This confused me for a second but I suppose it matches what the non chunked version does. The `.uploadSymbols` endpoint returns a `CompletableFuture<Void>` because there is no response body.

I'd suggest adding a small comment here to explain that in case my future self gets confused again :D.